src/box/xlog_reader.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2025, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#pragma once
+
+#include "salad/stailq.h"
+#include "xrow.h"
+#include "error.h"
+
+#if defined(__cplusplus)
+extern "C" {
+#endif /* defined(__cplusplus) */
+
+/**
+ * A request that xlog can hold with some technical fields for
+ * `xlog_reader`.
+ */
+struct recovery_request {
```

Comment:
Probably better name it `xlog_entry` and put it in `xlog.h` (if we agree to implement synchronous xlog reader there).